src/OVAL/probes/unix/process58.c

```diff
@@ -351,6 +352,83 @@ static int get_exec_shield_status(int pid) {
 	return ret;
 }
 
+/**
+ * Parse /proc/%d/cmdline file
+ * @param filepath Path to file ~ use preallocated buffer for the path
+ * @param buffer output buffer with non-zero size
+ * @return ps-like command info or NULL
+ */
+static inline bool get_process_cmdline(const char* filepath, struct oscap_buffer* const buffer){
+
+	int fd = open(filepath, O_RDONLY, 0);
+
+	if (fd < 0) {
+		return false;
+	}
+
+	oscap_buffer_clear(buffer);
+
+
+
+	for(;;) {
+		static const int chunk_size = 1024;
+		char chunk[chunk_size];
+		// Read data, store to buffer
+		ssize_t read_size = read(fd, chunk, chunk_size );
+		oscap_buffer_append_binary_data(buffer, chunk, read_size);
+
+		// If reach end of file, then end the loop
+		if (chunk_size != read_size) {
+			break;
+		}
+	}
+
+	close(fd);
+
+	int length = oscap_buffer_get_length(buffer);
+	char* buffer_mem = oscap_buffer_get_raw(buffer);
+
+	if ( length == 0 ) { // empty file
+		return false;
+	} else {
+		/**
+		 * Skip multiple trailing zeros
+		 */
+		int i = length - 1;
+		while ( (i > 0) && (buffer_mem[i] == '\0') ) {
+			--i;
+		}
+
+		/**
+		 * Program and args are separated by '\0' and '\n'
+		 * Replace them with spaces ' '
+		 */
+		while( i >= 0 ){
+			char chr = buffer_mem[i];
+			if ( ( chr == '\0') || ( chr == '\n' ) ) {
+				buffer_mem[i] = ' ';
+			} else if ( chr == '\033' ) {
```

Comment:
This really needs a comment. Is the `escape` char the only thing we need to handle here?